### DISCUSSION TITLE
Bubble Tea with msys2/mintty?

### DISCUSSION THREAD

**Discussion by unreal** [charmbracelet/bubbletea#830] [Q&A]
Hi there,

I typically use msys2[1] for a development environment on Windows. Unfortunately Bubble Tea does not seem to work in this environment... here is the result of running the README tutorial:

![image](https://github.com/charmbracelet/bubbletea/assets/9277/079e532a-7de6-486f-a516-e84ab84392f0)

As you can see, any key presses simply get written to the terminal and don't make it to the app process. Arrow keys seem to move the terminal cursor (not the Bubble Tea cursor). Things work just find in a Windows cmd prompt.

This is almost certainly an issue with my own msys2/mintty/bash environment, but I was wondering if anyone has experienced this issue and/or has any advice on fixing the environment to work with Bubble Tea.

The only discussion I could find semi-related to this is here[2].

Thank you!

[1]: https://www.msys2.org/
[2]: https://github.com/charmbracelet/bubbletea/discussions/312

**Comment by bashbunni** [comment]
Hey @unreal it looks like this might be a bug in [termenv](https://github.com/muesli/termenv/issues/77) that is making mintty not work properly with Bubble Tea. Sorry I don't have any immediate solution here

**Comment by meowgorithm** [comment]
Hey @unreal! To provide some more context per Bash's response, we haven't spent much time with msys2, so Bubble Tea support is probably lacking there. Not knowing much about the platform yet, I can't say whether or not we're able to support it, but in the meantime I can confidently say Bubble Tea support for both the Windows Command Console and WSL is quite good.
